### DISCUSSION TITLE
Grove - DS1307 RTC (Real Time Clock) for Arduino | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#46] [Q&A]
# Grove - DS1307 RTC (Real Time Clock) for Arduino | Seeed Studio Wiki

Grove - DS1307 RTC (Real Time Clock) for Arduino

https://wiki.seeedstudio.com/Grove-RTC/

**Comment by D4H13N** [comment]
Big problems :
error: no matching function for call to 'TwoWire::begin(uint16_t&, uint16_t&)'
     Wire.begin(SDA,SCL);
note: candidate: void TwoWire::begin()
     void begin();
note:   candidate expects 0 arguments, 2 provided
note: candidate: void TwoWire::begin(uint8_t)
     void begin(uint8_t);
note:   candidate expects 1 argument, 2 provided
note: candidate: void TwoWire::begin(int)
     void begin(int);
note:   candidate expects 1 argument, 2 provided

exit status 1

Compilation error: exit status 1

**Reply by MatthewJeffson** [reply]
Hi, which board did you use and would you mind send the detail to the [forum](https://forum.seeedstudio.com/)?

**Reply by D4H13N** [reply]
Hi MatthewJeffson,
Board : UnoR3
Software : Arduino IDE 2.0.4 | CLI Version 0.31.0
Libraries from this page, the code too.
Do you need other information?

**Reply by MatthewJeffson** [reply]
Hi! that will be all, thanks!
Hope that there will be some of our friends can help you!
Hope you have a good day!

**Reply by benson225** [reply]
Hi,

I am having the same issues when using the same Arduino also, wondering if you ever managed to solve this?

Thanks

**Reply by MatthewJeffson** [reply]
Hi, I am sorry the inconvenience that our tech support platforms([forum](https://forum.seeedstudio.com/), [email](https://www.seeedstudio.com/contacts) (techsupport@seeed.io), [discord](https://discord.com/invite/eWkprNDMU7)) are presented somewhere else.

We are trying to enhance the presentation of the overall wiki framework first and we are running the [contributor project](https://github.com/orgs/Seeed-Studio/projects/6?pane=issue&itemId=30957479) here for asking collaborators for help. 
We really welcome your thoughts on updates to our wiki platform.

Thanks!

**Reply by rootvoid32** [reply]
so to solve the problem do this
https://www.arduino.cc/reference/en/libraries/grove-rtc-ds1307/
go here and download the libary
after go line 55 DS1307 on notepad++
and put this TwoWire.begin(SDA,SCL); instead of that Wire.begin(SDA,SCL);

void

**Comment by edinel** [comment]
So with an Arduino Nano IOT33, on a Grove Shield, hooked up to the I2C port, the output is not at all what it's supposed to be.:

45:165:85	165/165/2165 165* 
45:165:85	165/165/2165 165* 
45:165:85	165/165/2165 165* 
45:165:85	165/165/2165 165* 
45:165:85	165/165/2165 165* 
45:165:85	165/165/2165 165*

This output doesn't make sense, of course, and the code is exactly the example.  
{
	Serial.begin(9600);
	clock.begin();
	clock.fillByYMD(2023,4,30);//Jan 19,2013
	clock.fillByHMS(17,45,30);//15:28 30"
	clock.fillDayOfWeek(SUN);//Saturday
	clock.setTime();//write time to the RTC chip
}

**Reply by xzxcessarr** [reply]
Hello, from your description we suppose maybe there is some error in RTC crystal or somewhere. You can test by the list here:
1. check whether you are using the I2C interface to connect with your RTC
2. use the jumper to connect the module to your I2C interface on your board
3. check the battery is well or not
Thanks!

**Reply by edinel** [reply]
It turns out that this WAS a battery issue!  Thank you!

**Comment by MatthewJeffson** [comment]
Welcome to our product documentation platform! We value user feedback and contributions, so we couldn’t be more grateful that you can help us improving the documentation.

Hence, if you find obvious errors in the product documentation, or you have additional ideas and insights about using products, please [contribute](https://wiki.seeedstudio.com/Contribution-Guide) to us.

If you need technical support, you can email it to [fae@seeed.cc](mailto:fae@seeed.cc).

If you want to discuss the product, I recommend the [Seeed Studio Forum](https://forum.seeedstudio.com/) and [Seeed Studio Discord Channel](https://discord.gg/eWkprNDMU7)

The "Discussion channel" in Github here is a relatively new addition and may not receive much attention.

Nonetheless, we hope you find it helpful with the help of others. We will clean it up on a monthly basis.

**Comment by rootvoid32** [comment]
so to solve the problem do this
 https://www.arduino.cc/reference/en/libraries/grove-rtc-ds1307/
go here and download the libary 
after go line 55 DS1307 on notepad++
and put this  TwoWire.begin(SDA,SCL); instead of that Wire.begin(SDA,SCL);

void
